src/test/java/com/epam/aidial/core/ResourceBaseTest.java

```diff
@@ -211,12 +213,35 @@ static void verify(Response response, int status, String body) {
     static void verifyJson(Response response, int status, String body) {
         assertEquals(status, response.status());
         try {
-            assertEquals(ProxyUtil.MAPPER.readTree(body).toPrettyString(), ProxyUtil.MAPPER.readTree(response.body()).toPrettyString());
+            JsonNode expected = sortPermissions(ProxyUtil.MAPPER.readTree(body));
+            JsonNode actual = sortPermissions(ProxyUtil.MAPPER.readTree(response.body()));
+            sortPermissions(actual);
+            assertEquals(expected.toPrettyString(), actual.toPrettyString());
         } catch (JsonProcessingException e) {
             throw new RuntimeException(e);
         }
     }
 
+    static void verifyMetadata(MetadataBase expected, String actual) throws JsonProcessingException {
+        assertEquals(
+                sortPermissions(ProxyUtil.MAPPER.readTree(ProxyUtil.MAPPER.writeValueAsString(expected))),
+                sortPermissions(ProxyUtil.MAPPER.readTree(actual)));
+    }
+
+    private static JsonNode sortPermissions(JsonNode root) {
```

Comment:
this method is not generic, I would add permissions in a right order or think about generic way to deal with set-like structures

Reply:
In JSON, either sets or lists are represented identically. The order of elements in sets is usually not guaranteed (unless it's a special type of set) and may change from run to run.